packages/@aws-cdk/aws-apigateway/lib/method.ts

```diff
@@ -364,6 +367,100 @@ export class Method extends Resource {
     // For backward compatibility
     return options.requestValidator?.requestValidatorId;
   }
+
+  /**
+   * Returns the given named metric for this API method
+   */
+  public metric(metricName: string, stage: IStage, props?: cloudwatch.MetricOptions): cloudwatch.Metric {
```

Comment:
I find this confusing; the doc string says that it returns "the given named metric", as if the metric already exists, but then the method invokes a constructor.

can we rename this to or `attachMetric` and change the docstring to something like "adds the named metric for this API method. It has metric name `metricName` and dimensions: 

`{please list them here in some form}`

to this method."

Reply:
I've mainly copied the existing code from `restapi.ts`, all the metric methods invoke a constructor and return a metric after calling `attachTo()`.

The docstring seems to match metrics in other modules such as SQS and Lambda

I would think that renaming the docstring to something like "Returns a metric for this API method with the given name" would be sufficient to make it clearer